### ISSUE TITLE
Complete DSP API

### ISSUE THREAD

**Issue by james34602** [maxmpz/powerampapi#19]
Currently the DSP API is not fully complete, I would like to my JamesDSP to PowerAMP enable no root DSP.
Unfortunately the current API is allow me to that.
I would like have some mandatory function about it, for example, initialization(Include information about buffer length, channel count, sample rate, data type, bit depth), and buffer receive function call and DSP closing finalization.

**Comment by maxmpz** [comment]
Right, docs and working sample project will be updated/added in the upcoming weeks. 
Generally it's possible to load own .so into Poweramp process and have own plugin UI in a separate process.
